### ISSUE TITLE
[Feature Request] Avoid fetching attachments if it's already downloaded

### ISSUE THREAD

**Issue by yrccondor** [barbushin/php-imap#585]
I'm using this library for a while, and it's really helpful!

I noticed that when I dealing with mails with multiple inline iamges (cid:...), `IncommingMail::embedImageAttachments()` took a very long time to process (I've set `$attachmentsDir` to `null`). I found that for 
every call to `IncomingMailAttachment::getContents()` will eventually call `imap_fetchbody` in `Imap::fetchbody()`, which took a long time to fetch.

Since we already have attachment data in `DataPartInfo::data`, can we just return it in `DataPartInfo::fetch()` if data is set or add a getter for `DataPartInfo::data`?

I'm not really familiar with imap, sorry if I'm wrong.

**Comment by yrccondor** [comment]
I've made a quick fix. This works for me.

`DataPartInfo::fetch()`:

```php
public function fetch(): string
{
    if (0 === $this->part) {
        $this->data = Imap::body($this->mail->getImapStream(), $this->id, $this->options);
    } else {
        if (null !== $this->data) {
            return $this->data;
        }
        $this->data = Imap::fetchbody($this->mail->getImapStream(), $this->id, $this->part, $this->options);
    }

    return $this->decodeAfterFetch();
}
```

BTW, since only a "Content-ID" is also enough:

https://github.com/barbushin/php-imap/blob/e9339769abed0a63ef91aedf200e118e1b41dd99/src/PhpImap/IncomingMail.php#L222-L226

It's may better to change this line to `if ($attachment->contentId == $cid) {`, otherwise in some cases we'll get a wrong image.

**Comment by sunnyzss928** [comment]
The pictures and attachments in my mail cannot be displayed. Why? The picture address is shown as cid:dhhdudjdhefbjujfddfwe.jpg
But it cannot be loaded.
Do I need to replace the path? Is there a problem in that place?

**Comment by Sebi94nbg** [comment]
Thank you very much for this code improvement @yrccondor!

I've implemented and tested it.

**Comment by Sebi94nbg** [comment]
This feature/change has been released in version [4.5.0](https://github.com/barbushin/php-imap/releases/tag/4.5.0).
